File: boj7490_0만들기.java
import java.io.*;
import java.util.*;
import java.math.*;

public class boj7490_0만들기 {
    static int n;
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static char[] numbers = new char[9];
    static char[] symbols = new char[3];
    static StringBuilder sb = new StringBuilder();
    static int ans = 0;

    public static int eval() throws IOException {
        int idx = 0;
        int num = 0;
        ArrayList<Character> quiz = new ArrayList<Character>();
        ArrayList<Integer> nums = new ArrayList<Integer>();
        
        int sum = 0;

        quiz.add(numbers[0]);
        for (int i = 0; i < n-1; ++i) {
            quiz.add(sb.charAt(i));
            quiz.add(numbers[i+1]);
        }
        for (int i = 0; i < n-1; ++i) {
            num += Character.getNumericValue(numbers[i]);
            if (sb.charAt(i) == '+' || sb.charAt(i) == '-') {
                nums.add(num);
                num = 0;
            } else if (sb.charAt(i) == ' ') {
                num *= 10;
            }
        }
        num += Character.getNumericValue(numbers[n-1]);
        nums.add(num);


        sum = nums.get(0);
        idx = 0;
        for (int i = 0; i < n -1; ++i) {
            if (sb.charAt(i) == ' ')
                continue;
            if (sb.charAt(i) == '+') {
                sum += nums.get(idx+1);
            } else if (sb.charAt(i) == '-') {
                sum -= nums.get(idx+1);
            }
            ++idx;
        }
        if (sum == 0) {
            // System.out.println(sum);
            for (int i = 0; i < quiz.size(); ++i) {
                bw.write(quiz.get(i));
            }
            bw.write("\n");
        }
        return sum;
    }

    public static void dfs(int depth) throws IOException {
        if (depth >= n - 1) {
            // System.out.println(sb);
            eval();
            return ;
        }
        for (int i = 0; i < 3; ++i) {
            sb.append(symbols[i]);
            dfs(depth + 1);
            sb.deleteCharAt(sb.length() - 1);
        }
    }
    public static void main(String[] args) throws IOException {
        symbols[0] = ' ';
        symbols[1] = '+';
        symbols[2] = '-';
        for (int i = 0; i < 9; ++i) {
            int c = i+1 + 48;
            numbers[i] = (char)c;
        }
        int t = Integer.parseInt(br.readLine());
        while (t > 0) {
            n = Integer.parseInt(br.readLine());
            dfs(0);
            bw.write("\n");
            --t;
        }        
        bw.flush();
        bw.close();
    }
}

File: boj7490_0만들기.kt
class Boj7490 {
    val n = readln().toInt()
    val numbers = CharArray(9) { (it+1 + '0'.code).toChar() }
    val symbols = charArrayOf(' ', '+', '-');
    val list = mutableListOf<Char>();

    fun dfs(depth: Int) {
        if (depth >= n-1) {
            val quiz = mutableListOf<Char>()
            quiz += '1'
            for (i in list.indices) {
                quiz += list[i]
                quiz += numbers[i+1]
            }

            val numList = mutableListOf<Int>()
            var num = 0
            for (i in quiz.indices) {
                if (quiz[i] != '+' && quiz[i] != '-' && quiz[i] != ' ') {
                    num += quiz[i].digitToInt()
                } else if (quiz[i] == ' ') {
                    num *= 10
                } else {
                    numList += num
                    num = 0
                }
            }
            numList += (num)
            var sum = numList[0]
            var idx = 0
            for (i in list.indices) {
                if (list[i] == ' ')
                    continue
                if (list[i] == '-') {
                    sum -= numList[idx+1]
                } else if (list[i] == '+') {
                    sum += numList[idx+1]
                }
                ++idx
            }
            if (sum == 0) {
                println(String(quiz.toCharArray()))
            }
            // println(numList)
            // println(sum)
            return
        }
        for (i in symbols.indices) {
            list += symbols[i]
            dfs(depth+1)
            list.removeLast()
        }
    }

    fun solution() {
        dfs(0)
    }
}

fun main() {
    repeat(readln().toInt()) {
        Boj7490().solution()
        println()
    }
}